### ISSUE TITLE
loss 计算疑问

### ISSUE THREAD

**Issue by sunnysky29** [jingyaogong/minimind#335]
https://github.com/jingyaogong/minimind/blob/d9453ed9a35847d7363a4b16b8a84274941c3609/train_pretrain.py#L54
                Y.view(-1)
            ).view(Y.size())
            loss = (loss * loss_mask).sum() / loss_mask.sum()
            loss += res.aux_loss
            loss = loss / args.accumulation_steps

这里梯度累计时候，为何 loss 要除以accumulation_steps 呢？？
必须这么写么，有没有特殊情况？

**Comment by JimmyAwoe** [comment]
这里应该是用到了一种叫做梯度积累（gradient accumulation）的技术。

大体思路是每次只计算batch的一部分，计算出来loss后存储起来。直到计算的数量达到所需的batch数量时，才进行一次更新。这么做的目的主要是为了节省运行时的内存要求，每次只需要加载很少的数据集到内存就可以完成训练。
